src/main/java/cubicchunks/server/PlayerCubeMap.java

```diff
@@ -228,57 +243,108 @@ public void tick() {
 		//sort toLoadPending if needed, but at most every 4 ticks
 		if (this.toGenerateNeedSort && currentTime%4L == 0L) {
 			this.toGenerateNeedSort = false;
-			Collections.sort(this.toGenerate, (watcher1, watcher2) ->
+			Collections.sort(this.cubesToGenerate, (watcher1, watcher2) ->
 					ComparisonChain.start().compare(
 							watcher1.getClosestPlayerDistance(),
 							watcher2.getClosestPlayerDistance()
 					).result());
+			// We don't sort columns because we generate all columns
 		}
 		getWorld().getProfiler().endStartSection("sortToSend");
-		//sort toSendToClient every other 4 ticks
+		//sort cubesToSendToClients every other 4 ticks
 		if (this.toSendToClientNeedSort && currentTime%4L == 2L) {
 			this.toSendToClientNeedSort = false;
-			Collections.sort(this.toSendToClient, (watcher1, watcher2) ->
+			Collections.sort(this.cubesToSendToClients, (watcher1, watcher2) ->
 					ComparisonChain.start().compare(
 							watcher1.getClosestPlayerDistance(),
 							watcher2.getClosestPlayerDistance()
 					).result());
+			// We don't sort columns because we send all columns
 		}
 
 		getWorld().getProfiler().endStartSection("generate");
-		if (!this.toGenerate.isEmpty()) {
+		if (!this.columnsToGenerate.isEmpty()) {
+			getWorld().getProfiler().startSection("columns");
+			Iterator<PlayerCubeMapColumnEntry> iter = this.columnsToGenerate.iterator();
+			while (iter.hasNext()) {
+				PlayerCubeMapColumnEntry next = iter.next();
+
+				getWorld().getProfiler().startSection("column[" + next.getPos().chunkXPos + "," + next.getPos().chunkZPos + "]");
+				boolean success = next.getColumn() != null;
+				if (!success) {
+					boolean canGenerate = next.hasPlayerMatching(CAN_GENERATE_CHUNKS);
+					getWorld().getProfiler().startSection("generate");
+					success = next.providePlayerChunk(canGenerate);
+					getWorld().getProfiler().endSection(); // generate
+				}
+
+				if (success) {
```

Comment:
`providePlayerChunk` claims it can fail, by virtue of returning a boolean. Not sure in which cases that happens - maybe when async loading hits the load couldn't be finished yet at that point.